build/rust/config.gni

```diff
@@ -13,6 +13,7 @@ import("//build/config/sysroot.gni")
 declare_args() {
   rustup_home = "//brave/build/rustup/$brave_rust_version"
   cargo_profile = ""
+  enable_rust_lto = false
```

Comment:
mac/ios was the only platform we enabled this for and it was specifically to deal with the "too many personalities" issue, but actually I wonder if we should be using rust LTO on other platforms to reduce binary size. Since they export so much stuff, it might not be getting stripped in the final output. macos/ios are ok because we use https://github.com/brave/brave-core/pull/16789/files#diff-84fabac7eb11280108df0ac0414a13e868865f07329ae57baff447d9b477c0deR20 to strip everything

Reply:
I think it's a separate issue from this PR though because it doesn't change anything for other platforms